tests/api/shelves/by_ids.test.js

```diff
@@ -19,72 +21,140 @@ describe('shelves:by-ids', () => {
   })
 
   it('should be empty when the id does not exist', async () => {
-    const res = await authReq('get', `${endpoint}&ids=00000000000000000000000000000000`)
-    res.shelves.should.deepEqual({})
+    await publicReq('get', `${endpoint}&ids=${someCouchUuid}`)
+    .then(shouldNotBeCalled)
+    .catch(err => {
+      err.statusCode.should.equal(404)
+    })
   })
 
-  it('should get a public shelf', async () => {
-    const { shelf } = await createShelf()
-    shelf.listing.should.equal('public')
-    const res = await publicReq('get', `${endpoint}&ids=${shelf._id}`)
-    res.shelves.should.be.ok()
+  describe('visibility:public', () => {
+    it('should get a public shelf', async () => {
+      const { shelf } = await createShelf()
+      shelf.visibility.should.deepEqual([ 'public' ])
+      const { shelves } = await publicReq('get', `${endpoint}&ids=${shelf._id}`)
+      shelves[shelf._id].should.be.an.Object()
+    })
   })
 
-  it('should not return non friends network shelves', async () => {
-    const { shelf } = await createShelf(null, { listing: 'private' })
-    const res = await authReqB('get', `${endpoint}&ids=${shelf._id}`)
-    const resIds = Object.keys(res.shelves)
-    resIds.should.not.containEql(shelf._id)
-  })
+  describe('visibility:private', () => {
+    it('should return owner private shelves', async () => {
+      const { shelf } = await createShelf(null, { visibility: [] })
+      const res = await authReq('get', `${endpoint}&ids=${shelf._id}`)
+      const resIds = Object.keys(res.shelves)
+      resIds.should.containEql(shelf._id)
+    })
 
-  it('should return owner private shelves', async () => {
-    const { shelf } = await createShelf(null, { listing: 'private' })
-    const res = await authReq('get', `${endpoint}&ids=${shelf._id}`)
-    const resIds = Object.keys(res.shelves)
-    resIds.should.containEql(shelf._id)
+    it('should not return a private shelf to an authentified user', async () => {
+      const { shelf } = await createShelf(null, { visibility: [] })
+      await authReqB('get', `${endpoint}&ids=${shelf._id}`)
+      .then(shouldNotBeCalled)
+      .catch(err => {
+        err.statusCode.should.equal(403)
+      })
+    })
+
+    it('should not return a private shelf to a friend', async () => {
+      const friend = await createUser()
+      const user = await getUser()
+      await makeFriends(friend, user)
+      const { shelf } = await createShelf(null, { visibility: [] })
+      await customAuthReq(friend, 'get', `${endpoint}&ids=${shelf._id}`)
+      .then(shouldNotBeCalled)
+      .catch(err => {
+        err.statusCode.should.equal(403)
+      })
+    })
+
+    it('should not return a private shelf to a non-authentified user', async () => {
+      const { shelf } = await createShelf(null, { visibility: [] })
+      await publicReq('get', `${endpoint}&ids=${shelf._id}`)
+      .then(shouldNotBeCalled)
+      .catch(err => {
+        err.statusCode.should.equal(401)
```

Comment:
It could be that it's mistakenly non-identified request or badly identified request (old cookies), while if properly identified, the user should have had access to that resource: 401 seems appropriate for that case. Furthermore, I don't think we should expect ids to be secret, as I don't have in mind a serious threat scenario in which that would be problematic